base/server/src/main/java/com/netscape/cmscore/base/ArgBlock.java

```diff
@@ -230,19 +214,18 @@ public BigInteger getValueAsBigInteger(String n)
             throws EBaseException {
         String v = (String) mArgs.get(n);
 
-        if (v != null) {
+        if (v == null) {
+            throw new EBaseException(CMS.getUserMessage("CMS_BASE_ATTRIBUTE_NOT_FOUND", n));
+        }
+        try {
+            return new BigInteger(v, 10);
+        } catch (NumberFormatException e) {
             try {
-                return new BigInteger(v, 10);
-            } catch (NumberFormatException e) {
-                try {
-                    return new BigInteger(v, 16);
-                } catch (NumberFormatException ex) {
-                    throw new EBaseException(
-                            CMS.getUserMessage("CMS_BASE_INVALID_ATTR_TYPE", n, ex.toString()));
-                }
+                return new BigInteger(v, 16);
+            } catch (NumberFormatException ex) {
+                throw new EBaseException(
+                        CMS.getUserMessage("CMS_BASE_INVALID_ATTR_TYPE", n, ex.toString()));
             }
-        } else {
-            throw new EBaseException(CMS.getUserMessage("CMS_BASE_ATTRIBUTE_NOT_FOUND", n));
         }
```

Comment:
This code can be flattened further like this:
```
try {
    return new BigInteger(v, 10);
} catch (NumberFormatException e) {
    // not decimal, check if it's hexadecimal
}

try {
    return new BigInteger(v, 16);
} catch (NumberFormatException e) {
    // not hexadecimal
    throw new BaseException(...);
}
```
I think this code actually has a problem. Suppose the value is "10" in hexadecimal, it will be processed incorrectly as 10 instead of 16.